### PR TITLE
Grab bag of small changes

### PR DESCRIPTION
**Pull request by notnoop**
Small independent changes:
* add tests to ensure nomad/api and nomad/structs are in sync
* deflake a rkt test and allow rkt driver to be built on macOS instances
* enable scripts linting in CI

### CODE REVIEW

File: api/structsync_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			var found structs.RestartPolicy
+			toStructs(t, &found, apiP)
+
+			require.EqualValues(t, c.expected, found)
+		})
+	}
+}
+
+func toStructsResource(t *testing.T, in *Resources) structs.Resources {
+	var out structs.Resources
```
Comment (dantoml) [inline api/structsync_test.go]: maybe flag these with `t.Helper()` for better error reporting?

File: drivers/rkt/api.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package rkt
+
+import (
+	"net"
+)
+
+type Pod struct {
```
Comment (dantoml) [inline drivers/rkt/api.go]: Can you include a package comment with a link to the original source for these? Will make updating them in the future easier.

File: drivers/rkt/driver_test.go [hunk 3]
```diff
@@ -412,23 +410,22 @@ func TestRktDriver_StartWaitRecoverWaitStop(t *testing.T) {
 	ch, err = harness.WaitTask(context.Background(), task.ID)
 	require.NoError(err)
 
-	wg.Add(1)
-	waitDone = false
 	go func() {
-		defer wg.Done()
-		result := <-ch
+		time.Sleep(300 * time.Millisecond)
```
Comment (schmichael) [inline drivers/rkt/driver_test.go]: Hopefully we can remove this sleep? If InspectTask can see a task then StopTask should be able to work without a sleep. Not sure why the original test had it. Perhaps ported from a time when such guarantees did not exist?

File: drivers/rkt/driver_test.go [hunk 4]
```diff
@@ -412,23 +410,22 @@ func TestRktDriver_StartWaitRecoverWaitStop(t *testing.T) {
...
+		require.NoError(d.StopTask(task.ID, 0, "SIGKILL"))
```
Comment (schmichael) [inline drivers/rkt/driver_test.go]: No t.Fatal calls in a goroutine.

File: drivers/rkt/driver_test.go [hunk 5]
```diff
@@ -412,23 +410,22 @@ func TestRktDriver_StartWaitRecoverWaitStop(t *testing.T) {
...
+
+	select {
+	case result := <-ch:
 		require.NoError(result.Err)
 		require.NotZero(result.ExitCode)
-		require.Equal(9, result.Signal)
-		waitDone = true
-	}()
+		// when killing a task, signal might not propagate
+		//require.Equal(9, result.Signal)
```
Comment (schmichael) [inline drivers/rkt/driver_test.go]: Huh, why did this change? Or was it always flaky?

Comment (notnoop) [reply drivers/rkt/driver_test.go]: This is was always flaky and it's due to getting the `wait: no child processes` error that gets us https://github.com/hashicorp/nomad/blob/master/drivers/shared/executor/executor_linux.go#L265-L266 .

File: api/structsync_test.go (hunk 1 above)
Comment (notnoop) [reply api/structsync_test.go]: I prefer having Helper() here - so if we have an internal conversion error, we can point to the line that generated.  I think `t.Helper()` makes more sense for helper functions that check equality where the exact line generating the error isn't useful.

File: drivers/rkt/driver_test.go (hunk 3 above)
Comment (notnoop) [reply drivers/rkt/driver_test.go]: removed it and it worked quite well.

**Pull request merged**